core/block_validator.go

```diff
@@ -32,16 +32,16 @@ import (
 // BlockValidator implements Validator.
 type BlockValidator struct {
 	config *params.ChainConfig // Chain configuration options
-	bc     *BlockChain         // Canonical block chain
+	hc     *HeaderChain        // Canonical header chain
 	engine consensus.Engine    // Consensus engine used for validating
 }
 
 // NewBlockValidator returns a new block validator which is safe for re-use
-func NewBlockValidator(config *params.ChainConfig, blockchain *BlockChain, engine consensus.Engine) *BlockValidator {
+func NewBlockValidator(config *params.ChainConfig, headerchain *HeaderChain, engine consensus.Engine) *BlockValidator {
```

Comment:
Can a `HeaderChain` exist without a nested block chain / block DB? Seeing block validator operate on `HeaderChain` makes me second guess this.

I thought we'd decided that:
1) `HeaderChain` would be called `BlockChain`'
2) `BlockChain` (as named today) would be called `BlockDatabase`
3) Possibly later on we have a lightweight "header only" chain interface called `HeaderChain`

Unless its possible to have a `HeaderChain` without block data, then I don't think the `HeaderChain` name makes sense.